ethereum/contracts/vault/Vault.sol

```diff
@@ -0,0 +1,93 @@
+// contracts/Vault.sol
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.0;
+
+import "../libraries/external/BytesLib.sol";
+import "./VaultGetters.sol";
+
+/**
+ * @title Vault
+ * @dev Main vault contract for verifying and processing VAAs
+ */
+contract Vault is VaultGetters {
+    using BytesLib for bytes;
+
+    /**
+     * @dev Event emitted when an emitter is registered
+     * @param chainId The chain ID of the emitter
+     * @param emitterAddress The emitter address as bytes32
+     */
+    event EmitterRegistered(uint16 indexed chainId, bytes32 emitterAddress);
+
+    /**
+     * @dev Constructor initializes parent VaultGetters
+     * @param wormholeAddr Address of the Wormhole contract
+     * @param chainId_ Chain ID for this vault
+     * @param evmChainId_ EVM Chain ID
```

Comment:
The `evmChainId_` should be the chain id of the vault correct? Perhaps it would be better to name these `aztecChainId` and `evmChainId`. Wdyt?